src/stream/src/executor/hash_agg.rs

```diff
@@ -199,6 +202,11 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
             &args.agg_calls,
             Some(&args.extra.group_key_indices),
         );
+        // NOTE: we assume the prefix of table pk is exactly the group key
+        let group_key_table_pk_projection = args.result_table.pk_indices()
+            [..args.extra.group_key_indices.len()]
+            .to_vec()
+            .into();
```

Comment:
Does it affect correctness? I mean if the watermark column is the second column of the group by keys.

Reply:
The projection is actually to handle the cases where watermark column is the second inside group key cols. See the following:

https://github.com/risingwavelabs/risingwave/blob/7defa87ff38bc715b8ebbb26408492a3a7ad5498/src/frontend/src/optimizer/plan_node/generic/agg.rs#L278-L329

When grouped by a window watermark column, the column will be added to the table schema in `group_key` bitset order but in another order in table pk. Executor assumes that the window column is the first col in table pk but doesn't assume the position of it in table schema.